### PR TITLE
Add new storage implementation: Couchbase

### PR DESCRIPTION
**Pull request by mstrYoda**
This pr aims to add new storage implementation of Couchbase database.

Thank you for your reviews 🙏

### CODE REVIEW

File: couchbase/couchbase.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+		},
+		Transcoder: gocb.NewLegacyTranscoder(),
+	})
+
+	if err != nil {
+		panic(err)
+	}
+
+	_, err = cb.Ping(&gocb.PingOptions{
+		Timeout: 5 * time.Second,
```
Comment (efectn) [inline couchbase/couchbase.go:35]: Why we use hardcoded timeout here?

File: couchbase/couchbase.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+	b := cb.Bucket(cfg.Bucket)
+
+	return &Storage{cb: cb, bucket: b}
+}
+
+func (s *Storage) Get(key string) ([]byte, error) {
+	out, err := s.bucket.DefaultCollection().Get(key, nil)
+	if err != nil {
+		return nil, err
```
Comment (efectn) [inline couchbase/couchbase.go:50]: Is it returning error when the key is not found? Returning nil seems better https://github.com/gofiber/storage/blob/main/storage.go#L10

File: couchbase/couchbase.go [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+func (s *Storage) Reset() error {
+	return s.cb.Buckets().FlushBucket(s.bucket.Name(), nil)
+}
+
+func (s *Storage) Close() error {
+	return s.cb.Close(nil)
+}
+
+func (s *Storage) Conn() *gocb.Cluster {
```
Comment (efectn) [inline couchbase/couchbase.go:84]: We can add simple testcase like https://github.com/gofiber/storage/blob/main/arangodb/arangodb_test.go#L136

**Review by efectn — COMMENTED** [review]
Don't forget to add dependabot, release drafter, test actions
Also edit security workflow

File: go.mod [hunk 4]
```diff
@@ -1,3 +1,25 @@
 module github.com/gofiber/storage
 
 go 1.17
+
+require (
```
Comment (gaby) [inline go.mod:5]: Why this did go.mod get modified? It should only be the one under `coundbase/go.mod`

File: couchbase/go.mod [hunk 5]
```diff
@@ -0,0 +1,12 @@
+module github.com/gofiber/storage/couchbase
+
+go 1.20
```
Comment (gaby) [inline couchbase/go.mod:3]: This should be `1.18`, like the official one https://github.com/couchbase/gocb/blob/master/go.mod

**Review by gaby — CHANGES_REQUESTED** [review]

File: couchbase/couchbase.go (hunk 2 above)
Comment (mstrYoda) [reply couchbase/couchbase.go:50]: That's appropriate way, it was in my mind I forgot to add 👍

File: go.mod (hunk 4 above)
Comment (mstrYoda) [reply go.mod]: It was modified unintentially. I rewrote the git commit history and force pushed new commit that removes modifications on base go.mod file 👍

**Comment by ReneWerner87** [comment]
@mstrYoda can you also update the other workflows(tests ...)

**Comment by ReneWerner87** [comment]
pls check gosec
image

**Comment by ReneWerner87** [comment]
> @mstrYoda can you also update the other workflows(tests ...)

adjustement in the test workflow is still missing

**Comment by ReneWerner87** [comment]
[changes.patch](https://github.com/gofiber/storage/files/11651367/changes.patch)
@mstrYoda pls add this patch

**Comment by mstrYoda** [comment]
> 



> > @mstrYoda can you also update the other workflows(tests ...)
> 
> adjustement in the test workflow is still missing



> [changes.patch](https://github.com/gofiber/storage/files/11651367/changes.patch) @mstrYoda pls add this patch

Hi @ReneWerner87 , I am preparing a full container test pipeline for this development, I will add it ASAP 👍

**Comment by ReneWerner87** [comment]
ok thx

**Comment by gaby** [comment]
@mstrYoda Any updates on this?

**Comment by mstrYoda** [comment]
> ok thx


> @mstrYoda Any updates on this?

Hi fellows 👋 

I completed the integration test workflow and manually checked it on my local docker env. Can you review it?

**Comment by gaby** [comment]
> > ok thx
> 
> > @mstrYoda Any updates on this?
> 
> Hi fellows 👋
> 
> I completed the integration test workflow and manually checked it on my local docker env. Can you review it?

There's a small merge conflict, before CI runs

**Comment by gaby** [comment]
@mstrYoda Something got messed up during the fix of conflicts, all the CI files have old stuff. Did you pull the latest master?

**Comment by mstrYoda** [comment]
> @mstrYoda Something got messed up during the fix of conflicts, all the CI files have old stuff. Did you pull the latest master?

Fixed 👍  I just include conflicted file from the IDE ui mistakenly 👍

File: .github/workflows/test-couchbase.yml [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+          docker run --name couchbase -d -p 8091-8097:8091-8097 -p 9123:9123 -p 11207:11207 -p 11210:11210 -p 11280:11280 -p 18091-18097:18091-18097 couchbase:enterprise-7.1.1
+          sleep 5
+          docker exec --tty couchbase couchbase-cli cluster-init -c localhost:8091 --cluster-username admin --cluster-password 123456 --cluster-ramsize 256 --services data
+          sleep 3
+          docker exec --tty couchbase couchbase-cli bucket-create -c localhost:8091 --username admin --password 123456 --bucket fiber_storage --bucket-type couchbase --bucket-ramsize 100
+      -   name: Install Go
+          uses: actions/setup-go@v4
+          with:
+            go-version: '${{ matrix.go-version }}'
+      -   name: Setup Golang caches
```
Comment (ReneWerner87) [inline .github/workflows/test-couchbase.yml:33]: not needed anymore

setup-go@v4 is handling the cache by default

**Review by gaby — CHANGES_REQUESTED** [review]

**Review by gaby — APPROVED** [review]

**Review by efectn — APPROVED** [review]

**Review by ReneWerner87 — APPROVED** [review]

**Pull request merged**